devices/libsdio/README.md

```diff
@@ -0,0 +1,277 @@
+|SDIO_EVENT_CARD_IRQ       | SDIO device requested an interrupt            |
+|SDIO_EVENT_RW_DONE        | transfer has been completed                   |
+|SDIO_EVENT_RW_WRITE_READY | host write buffer is ready                    |
+|SDIO_EVENT_RW_READ_READY  | host read buffer is ready                     |
+|SDIO_EVENT_CMD_DONE       | issued command has been completed             |
+|SDIO_EVENT_CMD_TIMEOUT    | issued command timed out                      |
+|SDIO_EVENT_CMD_CRC        | command redundancy check error ocurred        |
+|SDIO_EVENT_CMD_ENDBIT     | command end bit error occurred                |
+|SDIO_EVENT_CMD_INDEX      | issued command is not valid                   |
+|SDIO_EVENT_DATA_TIMEOUT   | data transfer timed out                       |
+|SDIO_EVENT_DATA_CRC       | data redundancy check error ocurred           |
+|SDIO_EVENT_DATA_ENDBIT    | data end bit error occurred                   |
+|SDIO_EVENT_DMA_BUFFER     | DMA buffer boundary encountered               |
+|SDIO_EVENT_DMA_ERROR      | a DMA error ocurred                           |
+|SDIO_EVENT_TUNING_RETUNE  | host controller signals the need for retuning |
+|SDIO_EVENT_TUNING_ERROR   | host controller tuning error                  |
+|SDIO_EVENT_BLOCK_GAP      | a transfer was interrupted                    |
+
+These defines are meant for usage with `sdio_eventEnable` and `sdio_eventRegister` functions exclusively.
+
+## API description
+
+Following sections describe the API in detail, including the behaviour of certain functions.
+
+---
+
+```c
+typedef enum {
+	sdio_read,
+	sdio_write
+} sdio_dir_t;
+```
+
+This enumeration type is meant as parameter for `sdio_transferDirect` and `sdio_transferBulk` functions to indicate the direction of a given data transfer.
+
+---
+
+```c
+typedef void (*sdio_event_handler_t)(void *arg);
+```
+
+This type describes a function pointer for an interrupt handler. Handlers of interrupts events are assigned to a specific event, and every event can only have a single handler. Handler argument is provided during handler registration using `sdio_eventRegister`, and will remain unchanged until the handler registration is modified via the same function.
+
+---
+
+```c
+int sdio_init(void);
+```
+
+This function initializes the SDIO inteface hardware and tries to detect and enable connected device. Should no device be present or it could not respond to SDIO initialization sequence, this function will return an error code and free unusued resources it acquired.
+
+> **NOTE:** This function should not be called multiple times. In order to reinitialize the interface, `sdio_free` should be called between concurrent `sdio_init` calls instead.
+
+Parameters:
+
+- none
+
+Returns:
+
+| code | description |
+|------|-------------|
+|  0   | success     |
+| -1   | failure     |
+
+> **NOTE:** Every failure is accompanied by explicit fault description printed into standard output.
```

Comment:
Side note - maybe it should print to stderr instead
```suggestion
> **NOTE:** Every failure is accompanied by explicit fault description printed onto standard output.
```

Reply:
No printing occurs anymore. These notes were deleted.